aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -417,7 +417,11 @@ Tensor linspace(
     int64_t steps,
     const TensorOptions& options) {
   TORCH_CHECK(steps >= 0, "number of steps must be non-negative");
-  Tensor result = at::empty({steps}, options);
+  auto result_options = options;
+  if (start.isComplex() || end.isComplex()){
```

Comment:
this behavior is not stated in the doc so doc should be updated as well. perhaps something on the lines:
```
dtype (torch.dtype, optional) – the desired data type of returned tensor. Default: if None, and start and end are real values, uses a global default (see torch.set_default_tensor_type()) else sets it to the complex type corresponding to the torch.get_default_tensor_type().
```
also as I mentioned in the issue, we should update the doc to state that start and can be `float` or `complex`

Reply:
Sure.